src/Migrations/DollarBraceEmbeddedVariableMigration.hack

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST\Migrations;
+
+use namespace Facebook\HHAST;
+
+final class DollarBraceEmbeddedVariableMigration extends StepBasedMigration {
+  private static function migrateLiteralExpression(
+    HHAST\LiteralExpression $node,
+  ): HHAST\LiteralExpression {
+    $children = $node->getChildren();
+    $child = \HH\Lib\C\first($children);
+    if (!($child is HHAST\NodeList<_>)) {
+      return $node;
+    }
+    $literal_parts = $child->getChildren();
+    $first_part = \HH\Lib\C\first($literal_parts);
+    if (
+      !(
+        $first_part is HHAST\DoubleQuotedStringLiteralHeadToken ||
```

Comment:
IStringLiteral

Reply:
I meant to exclude the other node types that implement `IStringLiteral` e.g. `SingleQuotedStringLiteralToken`